internal/access/identity.go

```diff
@@ -170,10 +170,6 @@ func UpdateIdentityInfoFromProvider(c RequestContext, oidc providers.OIDCClient)
 			logging.Errorf("failed to revoke invalid user session: %s", nestedErr)
 		}
 
-		if nestedErr := data.DeleteProviderUsers(db, data.ByIdentityID(identity.ID), data.ByProviderID(provider.ID)); nestedErr != nil {
```

Comment:
So previously we assume that if a sync fails then the user was removed from that IDP. Now we only invalid their access keys but don't assume the user was removed. Is that right? 

Now the only way to remove this relationship is when we delete the entire provider? 

I thought we needed some way to remove them on sync because they were removed from the provider.

Reply:
Its more likely that their session expired. With SCIM coming it makes more sense to rely on that for a user being deleted, rather than removing them when their session expires.

Its also still possible to remove them by deleting the user (rather than the whole provider). Their session becomes invalid upon being deleted from the provider, and they can no longer login.